.pipelines/ci.yml

```diff
@@ -36,161 +36,254 @@ variables:
+              sudo rpm -ivh https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm
+              sudo rpm --import https://packages.microsoft.com/keys/microsoft.asc
+              sudo dnf install -y https://packages.microsoft.com/config/rhel/8/packages-microsoft-prod.rpm
+              sudo dnf install -y azure-cli
+              sudo dnf install -y podman make
+              which podman
+              podman --version
+              whoami
+
+              # Source the podman control script
+              . ./hack/e2e/utils.sh
+              run_podman
+              validate_podman_running
+
+              # Run the build and capture the image tag
+              export PODMAN_REMOTE_ARGS="-r --url=tcp://127.0.0.1:8888"
+              IMAGE_TAG=$(make azext-aro | grep 'Successfully tagged' | awk '{print $3}' | cut -d':' -f2)
+              podman images
+
+              if [ -z "$IMAGE_TAG" ]; then
+                echo "Error: Failed to capture IMAGE_TAG."
+                exit 1
+              fi
+
+              # Retag the image with the ACR repository
+              podman tag localhost/azext-aro:${IMAGE_TAG} arosvcdev.azurecr.io/my-image-repo:${IMAGE_TAG}
+
+              # Push the image to ACR
+              podman push arosvcdev.azurecr.io/my-image-repo:${IMAGE_TAG}
+
+              # Optionally tag and push the image as 'latest'
+              podman tag arosvcdev.azurecr.io/my-image-repo:${IMAGE_TAG} arosvcdev.azurecr.io/my-image-repo:latest
+              podman push arosvcdev.azurecr.io/my-image-repo:latest
+
+              # Stop podman
+              kill_podman
+
+              # Clear Azure session AFTER all ACR operations are complete
+              az account clear
+            target: ubi8
+
+      - job: Lint_Az_ARO_Extension_1
+        dependsOn: Build_Test_And_Push_Az_ARO_Extension
+        pool:
+          name: 1es-aro-ci-pool
+        variables:
+          HOME: $(Agent.BuildDirectory)
+        steps:
+          - template: ./templates/template-checkout.yml
+          - script: |
+              set -xe
+              export AZDEV_CONFIG_DIR=$(Agent.BuildDirectory)/azdev-config
+              make test-python
+              [[ -z "$(git status -s)" ]]
+            target: python
+
+      - job: Build_Lint_Test_RP_And_Portal
+        dependsOn: Build_Test_And_Push_Az_ARO_Extension
+        pool:
+          name: 1es-aro-ci-pool
+        steps:
+          - template: ./templates/template-checkout.yml
+          - script: |
+              set -xe
+              export NO_CACHE=true
+
+              # Install podman
+              sudo rpm -ivh https://dl.fedoraproject.org/pub/epel-release-latest-8.noarch.rpm
+              sudo dnf install -y podman make
+              which podman
+              podman --version
+              whoami
+
+              # Start podman service
+              . ./hack/e2e/utils.sh
+              run_podman
+              validate_podman_running
```

Comment:
idea: these commands appear to be common. Can we extract them into a template so we can use them consistently, everywhere?

Reply:
I have created a dedicated template, template-podman.yml, which consolidates these commands, including the installation of Podman, starting the Podman service, and validation steps. This template is now used consistently across the pipeline.